src/modules/project/pages/projectView/views/projectMainBody/components/GoalModal.tsx

```diff
@@ -41,19 +44,30 @@ const denominationOptions = [
 export const GoalModal = ({ isOpen, onClose, goal, projectId, refetch, openDeleteModal }: Props) => {
   const { t } = useTranslation()
 
-  const { control, handleSubmit, loading, watch, errors, enableSubmit } = useProjectGoalForm(
+  const { control, handleSubmit, loading, watch, errors, enableSubmit, setValue } = useProjectGoalForm(
     goal || null,
     projectId,
     onClose,
     refetch,
   )
   const isCompleted = goal && goal.status === ProjectGoalStatus.Completed
 
+  const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false)
```

Comment:
for true false, scenario, you can simple use, useDisclosure, it give you {isOpen, onOpen, onClose, onToggle}. You don't have to handle it manually. It's okay this way too.